backend/test/integration/job_api_test.go

```diff
@@ -221,28 +226,47 @@ func (s *JobApiTestSuite) TestJobApis() {
 
 	// The scheduledWorkflow CRD would create the run and it synced to the DB by persistent agent.
 	// This could take a few seconds to finish.
-	// TODO: Retry list run every 5 seconds instead of sleeping for 40 seconds.
-	time.Sleep(40 * time.Second)
 
 	/* ---------- Check run for hello world job ---------- */
-	runs, totalSize, _, err := s.runClient.List(&runParams.ListRunsParams{
-		ResourceReferenceKeyType: util.StringPointer(string(run_model.APIResourceTypeEXPERIMENT)),
-		ResourceReferenceKeyID:   util.StringPointer(helloWorldExperiment.ID)})
-	assert.Nil(t, err)
-	assert.Equal(t, 1, len(runs))
-	assert.Equal(t, 1, totalSize)
-	helloWorldRun := runs[0]
-	s.checkHelloWorldRun(t, helloWorldRun, helloWorldExperiment.ID, helloWorldExperiment.Name, helloWorldJob.ID, helloWorldJob.Name)
+	if err := retrier.New(retrier.ConstantBackoff(40, time.Second), nil).Run(func() error {
```

Comment:
Should we follow the original to do item to retry every 5 seconds? Now it retries every second. In addition, do we need to retry for 40 times? I think we don't need so many retries.

Reply:
Sure, we can do that, I think the cost of trying to get it should be low so does it really matter?

Reply:
Had 40 times before to account for the previous 40 seconds of sleep, updated to 8 times and 5 seconds between to allow for roughly the same max time.

Reply:
Can you update this? Looks like it still retries 40 times?